src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -340,16 +336,46 @@ public async Task RemoveAccountAsync(ApplicationConfiguration appConfig, IAccoun
             }
         }
 
-        public Task<IReadOnlyList<IAccount>> GetAccountsAsync(
+        public async Task<IReadOnlyList<IAccount>> GetAccountsAsync(
             string clientID,
             string redirectUri,
             AuthorityInfo authorityInfo,
             ICacheSessionManager cacheSessionManager,
             IInstanceDiscoveryManager instanceDiscoveryManager)
         {
-            // runtime does not yet support account discovery
+            if (!_wamOptions.ListWindowsWorkAndSchoolAccounts)
+            {
+                _logger.Info("[WamBroker] Returning no accounts due to configuration option.");
+                return Array.Empty<IAccount>();
+            }
+
+            string correlationId = Guid.NewGuid().ToString();
+            List<IAccount> accounts = new List<IAccount>();
+            var requestContext = cacheSessionManager.RequestContext;
+            var instanceMetadata = await instanceDiscoveryManager.GetMetadataEntryAsync(authorityInfo, requestContext, false).ConfigureAwait(false);
```

Comment:
There is a better way of invoking this, it's `GetMetadataEntryTryAvoidNetworkAsync`. You'll have to just pass in all the environments from the all the accouints from the broker, something like 

```csharp
var envs = discoverAccountsResult.Accounts.Select(acc => acc.Environment).ToList();
var instanceMetadata = await instanceDiscoveryManager.GetMetadataEntryTryAvoidNetworkAsync(authorityInfo, envs, requestContext).ConfigureAwait(false);
```